Classes/Domain/AbstractAggregateRoot.php

```diff
@@ -58,34 +58,24 @@ protected function __construct(string $identifier)
     /**
      * Apply an event to the current aggregate root
      *
-     * If the event aggregate identifier and name is not set the event
-     * is automatically set with the current aggregate identifier
-     * and name.
+     * If the event is an instance of AggregateEventInterface, the aggregate identifier will be set
      *
      * @param EventInterface $event
-     * @param array $metadata
-     * @return void
      * @api
      */
-    final public function recordThat(EventInterface $event, array $metadata = [])
+    final public function recordThat(EventInterface $event)
```

Comment:
> That's true, but it usually happens right afterwards.

"Usually" is not good enough to go for such a concrete and explicit naming, that might "occassionally" be completely off ;) We should then just strip the `occuredAt` and replace it with `storedAt` or something. That will *always* be the true meaning of the timestamp then.

> I don't think that's related. The version is still controlled from the repository

I think we have a bit different opinion on the meaning of the aggregate version, see also further down where the discussion better fits. I better understand your point now, so I guess we can find a consens there.

> Again, if it's aggregate-relevant it should not be metadata IMO

Yeah, but I'm not talking Aggregate domain logic here, I'm talking of the infrastructural code of the Aggregate, i.e. `recordThat()`. IMO this is a valid point for the implementation to enrich the domain event with technical meta-data, which currently I'd see as `occuredAt` and `version`.
But if we do move the version out of the aggregate I'd be willing to get rid of the `occuredAt` (is there actually a use-case for having the exact occurence timestamp?) in favor of the `storedAt` inside the repository and make the aggregate more pure to the domain.